### ISSUE TITLE
Team Fortress 2 on Steam Deck freezes when activating voice chat

### ISSUE THREAD

**Issue by LennyMcLennington** [ValveSoftware/Source-1-Games#4313]
Problem can be reproduced by installing TF2 on a Steam deck and trying to use the in-game voice chat or the 'Test Microphone' button in the voice settings.

Your voice will be transmitted fine after the freezing stops, but every time you start talking it will freeze the game for some time.

If it's tested in Desktop Mode of the Steam Deck, then when you activate voice you'll probably see some popup related to the audio device changing or something like that? Not sure exactly what the popup indicates, or whether it's related to the game freezing, but it says the name of the audio device. Here's an example: 
![example](https://user-images.githubusercontent.com/28272981/192655720-fef16600-e30d-458b-a7da-eb7e7f886cdc.png)

**Comment by rabscootle** [comment]
Would your audio also make a really awful popping sound when you tried using it? I've had that happen alongside the freezes.

**Comment by LennyMcLennington** [comment]
> Would your audio also make a really awful popping sound when you tried using it? I've had that happen alongside the freezes.

No, but sometimes popping and crackling sounds just start happening randomly for me in SteamOS and rebooting usually fixes it.

**Comment by rabscootle** [comment]
Maybe it's a handheld mode issue that's exacerbated by Steam Input; You seem to be playing on Desktop in that img.

**Comment by LennyMcLennington** [comment]
> Interesting, maybe it's a handheld mode issue that's exacerbated by Steam Input; You seem to be playing on Desktop in that img.

I tested it in Desktop mode for troubleshooting purposes only, the experience was the same in both Desktop Mode and Gaming Mode, other than the "Raven/Raven2/FireFlight/Renoir Audio Processor Speaker" popup.

Also, I'm not sure what you mean when you say maybe it's exacerbated by Steam input. What do you mean by that?

**Comment by rabscootle** [comment]
> > Interesting, maybe it's a handheld mode issue that's exacerbated by Steam Input; You seem to be playing on Desktop in that img.
> 
> I tested it in Desktop mode for troubleshooting purposes only, the experience was the same in both Desktop Mode and Gaming Mode, other than the "Raven/Raven2/FireFlight/Renoir Audio Processor Speaker" popup. There wasn't really anything visual to show in Gaming Mode so I didn't screenshot it.
> 
> Also, I'm not sure what you mean when you say maybe it's exacerbated by Steam input. What do you mean by that?

Ahh disregard my replies then. I wasn't sure if the audio popping/crackling when attempting to use voice chat was possibly a bug in Handheld mode caused by the controller input software the Deck uses (steam input)
